sql_generators/derived_view_schemas/__init__.py

```diff
@@ -178,7 +178,13 @@ def generate(target_project, output_dir, parallelism, use_cloud_function):
     ]
 
     for dataset_path in dataset_paths:
-        view_directories = [path for path in dataset_path.iterdir() if path.is_dir()]
+        view_directories = [
+            path
+            for path in dataset_path.iterdir()
+            if path.is_dir()
+            and (path / VIEW_FILE).exists()
+            and not (path / SCHEMA_FILE).exists()
```

Comment:
I don't think view deploys update the schema from dry run if a schema.yaml exists, unlike table deploys.  So this might cause view deployment to publish outdated schemas and fail.  IMO we can update the schema if it already exists, but if not then any schemas that need to be updated should be included in this PR

Reply:
I think if there's an explicit `schema.yaml` file for a view it's reasonable to expect that to be kept up to date and be accurate (e.g. [your code for `View.view_schema`](https://github.com/mozilla/bigquery-etl/pull/5057/files#diff-3b6a7e291017a4af9ec3835836f0aac50418cd91772c9c649ddbbc69ab77ba2cR182-R183) seems to make that assumption).

Reply:
It's a safe assumption because those schemas get updated by this generator but that wouldn't be the case with this change.  Running this generator locally, at least `funnel_retention_clients` and `funnel_retention_week_4` have columns added so those would fail to publish here https://github.com/mozilla/bigquery-etl/blob/23fa414aa5f475f013ac5eecb6ab2839519f9f96/bigquery_etl/view/__init__.py#L371-L376
Those should be updated in this PR.  I would still lean towards updating existing schemas since we can't always rely on schemas being up to date which is the case for tables

Reply:
OK:
- I've changed the `View.view_schema` logic to always try dry-running first, then augment with the `schema.yaml` if one exists.
- I've changed the `derived_view_schemas` SQL generator to use `View.view_schema` to get schemas, rather than getting the deployed view schema with `Schema.for_table()` (which I suspect was probably preventing the logic you implemented in #5057 from working properly).
- I've changed the `derived_view_schemas` SQL generator to not skip views that already have `schema.yaml` files.